libsolidity/analysis/TypeChecker.cpp

```diff
@@ -76,13 +76,13 @@ bool TypeChecker::checkTypeRequirements(SourceUnit const& _source)
 	return Error::containsOnlyWarnings(m_errorReporter.errors());
 }
 
-TypePointer const& TypeChecker::type(Expression const& _expression) const
+Type const* const& TypeChecker::type(Expression const& _expression) const
```

Comment:
Can you check these cases, please? I think this should return `Type const*`.

Reply:
Yes, good point. I fixed the `const* const&` I saw, but the return value ones escaped my notice.
Now updated.